### PR TITLE
feat: support identify user by id

### PR DESCRIPTION
**Pull request by abdelhamid-f-nasser**
## Description of the change

Support the new API which allows identifying user with ID.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

Jira ID: MOB-13746

## Checklists

- [x] Check that the `uuid` is modified with the value passed to the `id` parameter when intercepting the request.

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by mnoseir** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  You have not included a CHANGELOG entry! 
You can find it at [CHANGELOG.md](https://github.com/Instabug/Instabug-React-Native/blob/master/CHANGELOG.md).
  </td>
    </tr>
  </tbody>
</table>


### Coverage Report
| Label | Coverage | Status |
| ----- | -------- | ------ |
| JavaScript | 96.2% | ✅ |
| Android | 49.2% | ✅ |
| iOS | 40.6% | ✅ |
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 845dffe87fcd9be50a32f11548f733f93d9e7f34
</p>

File: ios/RNInstabug/InstabugReactBridge.m [hunk 1]
```diff
@@ -191,8 +191,12 @@ - (dispatch_queue_t)methodQueue {
...
 }
 
-RCT_EXPORT_METHOD(identifyUser:(NSString *)email name:(NSString *)name) {
-    [Instabug identifyUserWithEmail:email name:name];
+RCT_EXPORT_METHOD(identifyUser:(NSString *)email name:(NSString *)name id:(nullable NSString *)id) {
+    [Instabug identifyUserWithID:id email:email name:name];
+}
+
+RCT_EXPORT_METHOD(identifyUserWithID:(NSString *)userID email:(NSString *)email name:(NSString *)name) {
+    [Instabug identifyUserWithID:userID email:email name:name];
```
Comment (a7medev) [inline ios/RNInstabug/InstabugReactBridge.m:199]: Is there a need for having both `identifyUserWithID` and `identifyUser`? As per our last discussion, I think we agreed on having the two scenarios handled in one method instead, please correct me if I'm wrong.

Also, did we verify that not passing the ID or passing it as `null`/`nil` would work fine?

File: src/modules/Instabug.ts [hunk 2]
```diff
@@ -196,15 +196,16 @@ export const setString = (key: StringKey, string: string) => {
...
 
 /**
- * Sets the default value of the user's email and hides the email field from the reporting UI
+ * Sets the default value of the user's email and ID and hides the email field from the reporting UI
  * and set the user's name to be included with all reports.
  * It also reset the chats on device to that email and removes user attributes,
  * user data and completed surveys.
  * @param email Email address to be set as the user's email.
  * @param name Name of the user to be set.
+ * @param id optional ID of the user to be set.
```
Comment (a7medev) [inline src/modules/Instabug.ts:205]:
```suggestion
 * @param [id] ID of the user to be set.
```

Comment (a7medev) [inline src/modules/Instabug.ts:205]: We can change `optional` to the JSDoc syntax of optional parameters `[id]`.

```suggestion
 * @param [id] ID of the user to be set.
```

File: android/src/test/java/com/instabug/reactlibrary/RNInstabugReactnativeModuleTest.java [hunk 3]
```diff
@@ -205,11 +205,26 @@ public void tearDown() {
...
+        // given
+
+        String email = "sali@instabug.com";
+        String userName = "salmaali";
+        String id = "salmaali";
+        // when
+        rnModule.identifyUser(email, userName, id);
+        // then
+        verify(Instabug.class,times(1));
+        Instabug.identifyUser(userName, email, id);
```
Comment (a7medev) [inline android/src/test/java/com/instabug/reactlibrary/RNInstabugReactnativeModuleTest.java:227]: I think let's refactor this test to be cleaner like we do with Flutter, the old format feels weird TBH.

Let's also rename the test to something better than `givenArgs$identifyUserwithId_whenQuery_thenShouldCallNativeApiWithArgs`

**Review by a7medev — CHANGES_REQUESTED** [review]
Good job @abdelhamid-f-nasser! Please check the comments added and we should be good to go 🙏🏼

File: ios/RNInstabug/InstabugReactBridge.m (hunk 1 above)
Comment (abdelhamid-f-nasser) [reply ios/RNInstabug/InstabugReactBridge.m:199]: > Is there a need for having both identifyUserWithID and identifyUser? As per our last discussion, I think we agreed on having the two scenarios handled in one method instead, please correct me if I'm wrong.
Yes you're right, I think I forgot to remove it from here.

> Also, did we verify that not passing the ID or passing it as null/nil would work fine?
Yes It's backwards compatible and works just fine.

File: examples/default/ios/InstabugTests/InstabugSampleTests.m [hunk 4]
```diff
@@ -188,8 +188,19 @@ - (void)testIdentifyUser {
...
-  OCMVerify([mock identifyUserWithEmail:email name:name]);
+  [self.instabugBridge identifyUser:email name:name id:nil];
+  OCMVerify([mock identifyUserWithID:nil email:email name:name]);
+}
+
+- (void)testIdentifyUserWithID {
+  id mock = OCMClassMock([Instabug class]);
+  NSString *email = @"em@il.com";
+  NSString *name = @"this is my name";
+  NSString *id = @"this is my id";
```
[the remark this answers is not in the document]
Comment (abdelhamid-f-nasser) [reply examples/default/ios/InstabugTests/InstabugSampleTests.m:199]: What do you think of userId ?

[the remark this answers is not in the document]
Comment (a7medev) [reply examples/default/ios/InstabugTests/InstabugSampleTests.m:199]: LGTM!

File: src/modules/Instabug.ts [hunk 5]
```diff
@@ -227,7 +228,7 @@ export const logUserEvent = (name: string) => {
 
 /**
  * Appends a log message to Instabug internal log.
- * These logs are then sent along the next uploaded report.
+ * These logs are then sent aalong the next uploaded report.
```
Comment (ahmedAlaaInstabug) [inline src/modules/Instabug.ts:231]: just a typo mistake

Comment (abdelhamid-f-nasser) [reply src/modules/Instabug.ts:231]: Nice catch @ahmedAlaaInstabug , thanks 🙏

File: ios/RNInstabug/InstabugReactBridge.h [hunk 6]
```diff
@@ -49,7 +49,7 @@
 
 - (void)setString:(NSString *)value toKey:(NSString *)key;
 
-- (void)identifyUser:(NSString *)email name:(NSString *)name;
+- (void)identifyUser:(NSString *)email name:(NSString *)name userId:(nullable NSString *)id;
```
Comment (a7medev) [inline ios/RNInstabug/InstabugReactBridge.h:52]: Let's change `id` here to `userId` as well.

```suggestion
- (void)identifyUser:(NSString *)email name:(NSString *)name userId:(nullable NSString *)userId;
```

**Review by a7medev — CHANGES_REQUESTED** [review]

File: ios/RNInstabug/InstabugReactBridge.h (hunk 6 above)
Comment (abdelhamid-f-nasser) [reply ios/RNInstabug/InstabugReactBridge.h]: Thanks for catching this 🙏

**Review by a7medev — APPROVED** [review]

**Pull request merged**